avro/src/main/scala/AvroRandomExtractor.scala

```diff
@@ -0,0 +1,49 @@
+package com.memsql.spark.examples.avro
+
+import collection.JavaConversions._
+import com.memsql.spark.etl.api.{UserExtractConfig, SimpleByteArrayExtractor}
+import com.memsql.spark.etl.utils.PhaseLogger
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+import org.apache.avro.Schema
+import org.apache.avro.generic.GenericData
+import org.apache.avro.io.{DatumWriter, EncoderFactory}
+import org.apache.avro.specific.SpecificDatumWriter
+
+import java.io.ByteArrayOutputStream
+
+class AvroRandomExtractor extends SimpleByteArrayExtractor {
+  var count: Int = 1
+  val parser: Schema.Parser = new Schema.Parser()
+  var generator: AvroRandomGenerator = null
+  var writer:DatumWriter[GenericData.Record] = null
+  var avroSchema: Schema = null
+
+  override def initialize(sparkContext: SparkContext, config: UserExtractConfig, batchInterval: Long, logger: PhaseLogger): Unit = {
+    val avroSchemaJson = config.getConfigJsValue("avroSchema") match {
+      case Some(s) => s
+      case None => throw new IllegalArgumentException("className must be set in the config")
+    }
+    count = config.getConfigInt("count").getOrElse(1)
+    avroSchema = parser.parse(avroSchemaJson.toString)
```

Comment:
Can this throw an exception?  If so, is the error message comprehensible, or should we wrap this and throw our own, more-helpful exception?